pkg/build/build.go

```diff
@@ -130,13 +131,30 @@ func (b *Builder) generateCombustionScript() error {
 	return nil
 }
 
-func (b *Builder) writeCombustionFile(contents string, filename string) error {
+func (b *Builder) writeBuildDirFile(contents string, templateData any, filename string) error {
+	destFilename := filepath.Join(b.eibBuildDir, filename)
+	return b.writeFile(contents, templateData, destFilename)
+}
+
+func (b *Builder) writeCombustionFile(contents string, templateData any, filename string) error {
 	destFilename := filepath.Join(b.combustionDir, filename)
-	err := os.WriteFile(destFilename, []byte(contents), os.ModePerm)
+	return b.writeFile(contents, templateData, destFilename)
+}
+
+func (b *Builder) writeFile(contents string, templateData any, filename string) error {
```

Comment:
The problem with this design is that standard files (where `templateData` is nil) will still go through the process of parsing content as a template and executing it.

Can we either restructure the `writeX()` functions or add a check for the `templateData` in `writeFile()` and only follow templating logic if we're dealing with a template?